geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,55 +8,234 @@
+      getPersistedDataStores
+    } catch {
+      case e: Exception => handleError(s"Error reading data stores:", e)
+    }
+  }
+
+  /**
+    * Deletes for blobs given id
+    */
+  delete("/blob/:alias/:id/?") {
+    val alias = params("alias")
+    blobStores.get(alias) match {
+      case None =>
+        BadRequest(reason = "AccumuloBlobStore is not initialized.")
+      case Some(abs) =>
+        val id = params("id")
+        logger.debug("Attempting to delete: {} from store: {}", id, alias)
+        abs.delete(id)
+        Ok(reason = s"deleted feature: $id")
+    }
+  }
+
+  /**
+    * Get for blobs given id
+    */
+  get("/blob/:alias/:id/?") {
+    val alias = params("alias")
+    blobStores.get(alias) match {
+      case None =>
+        BadRequest(reason = "AccumuloBlobStore is not initialized.")
+      case Some(abs) =>
+        val id = params("id")
+        logger.debug("Attempting to get blob for id: {} from store: {}", id, alias)
+        val (returnBytes, filename) = abs.get(id)
+        if (returnBytes == null) {
+          NotFound(reason = s"Unknown ID $id")
+        } else {
+          contentType = "application/octet-stream"
+          response.setHeader("Content-Disposition", "attachment;filename=" + filename)
+          Ok(returnBytes)
+        }
+    }
+  }
+
+  /**
+    * Post for blob binaries
+    */
+  post("/blob/:alias/?") {
+    val alias = params("alias")
+    blobStores.get(alias) match {
+      case None      =>
+        logger.error("AccumuloBlobStore is not initialized in BlobStore.")
+        BadRequest()
+      case Some(abs) =>
+        logger.debug("Attempting to ingest file to BlobStore")
+        try {
+          fileParams.get("file") match {
+            case None =>
+              logger.error("no file parameter in request")
+              BadRequest()
+            case Some(file) =>
+              val otherParams = collection.mutable.Map[String, String]()
+              multiParams.foreach{ case (s, p) => otherParams.update(s, p.head) }
+              if (!otherParams.contains(filenameFieldName)) {
+                // we put the true filename in here so that we can preserve it in the blob table
+                otherParams.put(filenameFieldName, file.getName)
+              }
+              val tempFile = File.createTempFile(UUID.randomUUID().toString, FilenameUtils.getExtension(file.getName))
+              val actRes = try {
+                file.write(tempFile)
+                abs.put(tempFile, otherParams.toMap) match {
+                  case Some(id) =>
+                    Created(body = id, headers = scala.collection.immutable.Map("Location" -> request.getRequestURL.append(id).toString))
+                  case None =>
+                    BadRequest(reason = "Unable to process file")
+                }
+              } catch {
+                case e: Exception => handleError("", e)
+              } finally {
+                tempFile.delete()
+              }
+              actRes
```

Comment:
i think you want to remove actRes - handle error will be the return type if the exception block gets triggered, but this would override that. So should be try { ... created or badrequest } catch { handle error } finally {} /end of method